lib/beaker/options/parser.rb

```diff
@@ -228,10 +228,22 @@ def parse_hosts_options
         hosts_options = Beaker::Options::HostsFileParser.parse_hosts_file(@options[:hosts_file])
         return hosts_options
       rescue Errno::ENOENT
-        require 'tempfile'
+        error_message = "\nHosts file '#{@options[:hosts_file]}' does not exist."
+        error_message << "\nTrying as beaker-hostgenerator input.\n\n"
+        $stderr.puts error_message
         require 'beaker-hostgenerator'
-        bhg_cli = BeakerHostGenerator::CLI.new( [ @options[:hosts_file] ] )
-        hosts_file_content = bhg_cli.execute
+
+        hosts_file_content = begin
+          bhg_cli = BeakerHostGenerator::CLI.new( [ @options[:hosts_file] ] )
+          bhg_cli.execute
+        rescue BeakerHostGenerator::Exceptions::Error,
+          BeakerHostGenerator::Exceptions::InvalidNodeSpecError => error
+          error_message = "\nbeaker-hostgenerator was not able to use this value as input."
+          error_message << "\nExiting with an Error.\n\n"
+          $stderr.puts error_message
+          raise error
```

Comment:
Is it common in Beaker code to print to STDERR *and* raise an exception?

How did you decide on doing both?  Not a problem, just curious if there's some funky reason for this.  As a library you usually take greater care over printing/logging/errors because you want to cede control to the application.